### PR TITLE
:bug: fix incorrect logr usage

### PR DESCRIPTION
**Pull request by interma**
Fix: https://github.com/kubernetes-sigs/controller-runtime/issues/187

### CODE REVIEW

File: pkg/source/source.go [hunk 1]
```diff
@@ -86,8 +86,8 @@ func (ks *Kind) Start(handler handler.EventHandler, queue workqueue.RateLimiting
 	i, err := ks.cache.GetInformer(ks.Type)
 	if err != nil {
 		if kindMatchErr, ok := err.(*meta.NoKindMatchError); ok {
-			log.Error(err, "if %s is a CRD, should install it before calling Start",
-				kindMatchErr.GroupKind)
+			log.Error(err, fmt.Sprintf("if %s is a CRD, should install it before calling Start",
```
Comment (mindprince) [inline pkg/source/source.go]: For better or for worse, this is specifically discouraged by logr. https://github.com/go-logr/logr/blob/9fb12b3b21c5415d16ac18dc5cd42c1cfdd40c4e/logr.go#L11

logr wants you to use structured logging.

Comment (interma) [reply pkg/source/source.go]: fixed, thanks.

File: pkg/source/source.go [hunk 2]
```diff
@@ -86,8 +86,8 @@ func (ks *Kind) Start(handler handler.EventHandler, queue workqueue.RateLimiting
 	i, err := ks.cache.GetInformer(ks.Type)
 	if err != nil {
 		if kindMatchErr, ok := err.(*meta.NoKindMatchError); ok {
-			log.Error(err, "if %s is a CRD, should install it before calling Start",
-				kindMatchErr.GroupKind)
+			log.Error(err, "Kind should be installed before calling Start",
```
Comment (DirectXMan12) [inline pkg/source/source.go]: please add the working back in as "if kind is a CRD, it should be installed before calling start"

Comment (interma) [reply pkg/source/source.go]: Fixed, thanks.

**Comment by interma** [comment]
looks failed travis blocked this PR.
@DirectXMan12  anything else needs me to do?
Thanks.

**Comment by DirectXMan12** [comment]
@interma can you rebase on the latest master? there was a pr up to fix that flake

**Comment by interma** [comment]
Sure, rebased.

**Pull request merged**